From a81c6d0634fbe0a4aa5836f277e4a8065d9a9b19 Mon Sep 17 00:00:00 2001
From: Jan-Marek Glogowski <glogow@fbihome.de>
Date: Sun, 9 Jan 2022 01:01:49 +0100
Subject: [PATCH] Qt use platform-independent visibility functions

show() isn't just some conveniently renamed setVisible(true), but
presents a window in a platform-dependant way, which - at least
for WASM - means maximized and without a title bar. This is not
really expected and WASM can also use multiple windows.

Change-Id: I3fbaec1dfc2cd32bf3a90a94dcee01ebc88e58c7
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/128179
Tested-by: Jenkins
Reviewed-by: Jan-Marek Glogowski <glogow@fbihome.de>
(cherry picked from commit f7346dc88200fa2160d39b4e4b4ae31fcad59a8c)
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/132624
Reviewed-by: Michael Weghorn <m.weghorn@posteo.de>
---
 vcl/qt5/QtFrame.cxx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vcl/qt5/QtFrame.cxx b/vcl/qt5/QtFrame.cxx
index 76e89ecdb76b..10b7e46909c5 100644
--- a/vcl/qt5/QtFrame.cxx
+++ b/vcl/qt5/QtFrame.cxx
@@ -461,7 +461,7 @@ void QtFrame::Show(bool bVisible, bool bNoActivate)
     if (!bVisible) // hide
     {
         pSalInst->RunInMainThread([this]() {
-            asChild()->hide();
+            asChild()->setVisible(false);
             if (m_pQWidget->isModal())
                 modalReparent(false);
         });
@@ -475,7 +475,7 @@ void QtFrame::Show(bool bVisible, bool bNoActivate)
         QWidget* const pChild = asChild();
         if (m_pQWidget->isModal())
             modalReparent(true);
-        pChild->show();
+        pChild->setVisible(true);
         pChild->raise();
         if (!bNoActivate && !isPopup())
         {
-- 
2.34.1

